src/clients/python/experimental_api_v2/library/httpclient.py

```diff
@@ -107,198 +104,447 @@ def close(self):
+        """
+        self._input['datatype'] = value
 
-        Raises
-        ------
-        InferenceServerException
-            If unable to unload the model.
+    @property
+    def shape(self):
+        """Get the shape of input associated with this object.
 
+        Returns
+        -------
+        list
+            The shape of input
         """
-        #response = self._client_stub.post("/api/modelcontrol/unload/" +
-        #                                       model_name)
-        #self._last_request_id = raise_if_error(response['NV-Status'])
-        raise_error('Not implemented')
+        self._input['shape']
+
+    @shape.setter
+    def shape(self, value):
+        """Sets the shape of input associated with this object.
+
+        Parameters
+        ----------
+        value : list
+            The shape of input
+        """
+        self._input['shape'] = value
+
+    def set_data_from_numpy(self, input_tensor):
+        """Set the tensor data (datatype, shape, contents) from the
+        specified numpy array for input associated with this object.
+
+        Parameters
+        ----------
+        input_tensor : numpy array
+            The tensor data in numpy array format
+        """
+        if not isinstance(input_tensor, (np.ndarray,)):
+            raise_error("input_tensor must be a numpy array")
+        self._input['datatype'] = np_to_triton_dtype(input_tensor.dtype)
+        self._input['shape'] = input_tensor.shape
+        # FIXMEV2 Use Binary data when available on the server.
+        self._input['data'] = [val.item() for val in input_tensor.flatten()]
+
+    def set_parameter(self, key, value):
+        """Adds the specified key-value pair in the requested input parameters
+
+        Parameters
+        ----------
+        key : str
+            The name of the parameter to be included in the request. 
+        value : str/int/bool
+            The value of the parameter
+        
+        """
+        if not type(key) is str:
+            raise_error(
+                "only string data type for key is supported in parameters")
+
+        self._parameters[key] = value
+
+    def clear_parameters(self):
+        """Clears all the parameters that have been added to the request.
+        
+        """
+        self._parameters.clear()
+
+    def _get_tensor(self):
+        """Retrieve the underlying input as json string.
+
+        Returns
+        -------
+        str
+            The underlying tensor as serialized json string
+        """
+        self._input['parameters'] = self._parameters
+        return self._input
+
+
+class InferOutput:
+    """An object of InferOutput class is used to describe a
+    requested output tensor for an inference request.
+
+    Parameters
+    ----------
+    name : str
+        The name of output tensor to associate with this object
+    """
+
+    def __init__(self, name):
+        self._output = {}
+        self._parameters = {}
+        self._output['name'] = name
```

Comment:
Rather than shoving in a dict, consider keeping `name` a string:
```
self._name = name
self._parameters = {}
```

Your `_get_tensor()` method becomes:
```
    return {'name': self._name, 'parameters': self._parameters}
```

Right now `_output` is a bit mushy with state.  It always has a `name` key and it might have a `parameters` key, but only if `_get_tensor()` was called.  And those parameters are also accessible via another member variable too.

Reply:
I agree. This class represents an output, so _name can be a "first class" member in the object. Same is true for other classes. Don't make a map to store member values, just store them directly.